modules/ingest-geoip/src/test/java/org/elasticsearch/ingest/geoip/MaxMindSupportTests.java

```diff
@@ -0,0 +1,481 @@
+                "A new Database type, "
+                    + databaseType
+                    + ", has been added, but this test has not been updated to know which fields we "
+                    + "do not support for it. Update TYPE_TO_UNSUPPORTED_FIELDS_MAP",
+                unsupportedFields
+            );
+            /*
+             * fieldNamesFromMaxMind:        The fields that MaxMind's Response object contains
+             * unusedFields:                 Fields that MaxMind's Response object contains that we do not claim to support
+             * unknownUnsupportedFieldNames: Fields that we document as unsupported that MaxMind's Response object doesn't even contain
+             * unknownSupportedFieldNames:   Fields that we document as supported that MaxMind's Response object doesn't even contain
+             */
+            final SortedSet<String> fieldNamesFromMaxMind = getFieldNamesUsedFromClass("", maxMindClass);
+            SortedSet<String> unusedFields = new TreeSet<>(fieldNamesFromMaxMind);
+            unusedFields.removeAll(supportedFields);
+            SortedSet<String> unknownUnsupportedFieldNames = new TreeSet<>(unsupportedFields);
+            unknownUnsupportedFieldNames.removeAll(unusedFields);
+            assertThat(
+                "We have documented fields in TYPE_TO_UNSUPPORTED_FIELDS_MAP that MaxMind does not actually support for "
+                    + databaseType
+                    + ": "
+                    + unknownUnsupportedFieldNames,
+                unknownUnsupportedFieldNames,
+                equalTo(Set.of())
+            );
+            assertThat(
+                "New MaxMind fields have been added for "
+                    + databaseType
+                    + " that we do not use or have documented in TYPE_TO_UNSUPPORTED_FIELDS_MAP. The actual list of fields is:\n"
+                    + getFormattedList(unusedFields),
+                unusedFields,
+                equalTo(new TreeSet<>(unsupportedFields))
+            );
+            SortedSet<String> unknownSupportedFieldNames = new TreeSet<>(supportedFields);
+            unknownSupportedFieldNames.removeAll(fieldNamesFromMaxMind);
+            assertThat(
+                "We are attempting to use fields that MaxMind does not support for " + databaseType + ": " + unknownSupportedFieldNames,
+                unknownSupportedFieldNames,
+                equalTo(Set.of())
+            );
+        }
+    }
+
+    public void testUnknownMaxMindResponseClassess() {
+        Set<Class<? extends AbstractResponse>> supportedMaxMindClasses = new HashSet<>(TYPE_TO_MAX_MIND_CLASS.values());
+        // First just a sanity check that there's no overlap between what's supported and what's not:
+        Set<Class<? extends AbstractResponse>> supportedAndUnsupportedMaxMindClasses = new HashSet<>(supportedMaxMindClasses);
+        supportedAndUnsupportedMaxMindClasses.retainAll(KNOWN_UNSUPPORTED_RESPONSE_CLASSES);
```

Comment:
I didn't realize that method existed! But I think I'll leave it as-is, because it's useful to be able to print out the set of things in the intersection in the error message.